contracts/amm/MinterAmm.sol

```diff
@@ -221,6 +226,25 @@ contract MinterAmm is
         _;
     }
 
+    /// @dev Prevents a contract from calling itself, directly or indirectly.
+    /// Calling a `nonReentrant` function from another `nonReentrant`
+    /// function is not supported. It is possible to prevent this from happening
+    /// by making the `nonReentrant` function external, and make it call a
+    /// `private` function that does the actual work.
+    modifier nonReentrant() {
```

Comment:
I remembered we talked about including the openzepline implementation or our own and I forget what conclusion we got to. Could you describe why we decided to implement our own version of nonReentrant when openzepline has their own that we could just us in the contract?

Reply:
good idea, done.